server/src/all-exceptions-filter.ts

```diff
@@ -10,19 +23,56 @@ export class AllExceptionsFilter implements ExceptionFilter {
     const response = ctx.getResponse();
     const request = ctx.getRequest();
 
-    const status = exception instanceof HttpException ? exception.getStatus() : HttpStatus.INTERNAL_SERVER_ERROR;
+    let errorResponse: ErrorResponse = { message: exception.message, status: HttpStatus.INTERNAL_SERVER_ERROR };
 
-    const message = exception?.response?.message || exception.message;
+    switch (exception.constructor) {
+      case HttpException:
+        errorResponse = { status: exception.getStatus(), message: exception?.response?.message || exception.message };
+        break;
+      case QueryFailedError:
+        errorResponse = this.handleQueryExceptions(exception);
+        break;
+      default:
+        break;
+    }
 
-    if (status === HttpStatus.INTERNAL_SERVER_ERROR) {
+    if (errorResponse.status === HttpStatus.INTERNAL_SERVER_ERROR) {
       this.logger.error(exception);
     }
 
-    response.status(status).json({
-      statusCode: status,
+    response.status(errorResponse.status).json({
+      statusCode: errorResponse.status,
       timestamp: new Date().toISOString(),
       path: request.url,
-      message,
+      message: errorResponse.message,
     });
   }
+
+  private handleQueryExceptions(exception: any): ErrorResponse {
+    const status = HttpStatus.UNPROCESSABLE_ENTITY;
+    const code = (exception as any).code;
+    let message: string = (exception as QueryFailedError).message;
+
+    switch (code) {
+      case PostgresErrorCode.UniqueViolation:
+        message = 'Already Existed!';
+        break;
+      case PostgresErrorCode.CheckViolation:
```

Comment:
Let's just raise Postgres error for CheckViolation, ForeignKeyViolation with 500. This can be checked for with validations at frontend/backend. Moreover, we would need to know of such exceptions to add in any such validation that was missed.